### ISSUE TITLE
conv1d precision error

### ISSUE THREAD

**Issue by jiqing-feng** [intel/torch-xpu-ops#1789]
### 🐛 Describe the bug

The output of conv1d is different with cuda/cpu

```python
import torch
from transformers.testing_utils import torch_device
 
 
inputs = torch.load("sat_hidden_states.pt", map_location=torch_device).transpose(1, 2)
weight = torch.load("sat_weight.pt", map_location=torch_device)
bias = torch.load("sat_bias.pt", map_location=torch_device)
 
output = torch.nn.functional.conv1d(inputs, weight, bias, dilation=1)
print(output)
```

CUDA/CPU output:

![Image](https://github.com/user-attachments/assets/a3249997-a37e-43d0-b427-fc286b9767b3)

XPU outputs:

![Image](https://github.com/user-attachments/assets/3a244b31-a0d3-4074-8a73-dd35a33b77d2)

The required tensors are attached.
[test_tensor.zip](https://github.com/user-attachments/files/20972956/test_tensor.zip)

### Versions

```
PyTorch version:   2.8.0.dev20250624+xpu
PyTorch CXX11 ABI: Yes
IPEX version:      N/A
IPEX commit:       N/A
Build type:        N/A

OS:                Ubuntu 22.04.5 LTS (x86_64)
GCC version:       (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Clang version:     N/A
IGC version:       2025.1.1 (2025.1.1.20250418)
CMake version:     version 3.22.1
Libc version:      glibc-2.35

Python version:    3.11.13 (main, Jun  4 2025, 08:57:29) [GCC 11.4.0] (64-bit runtime)
Python platform:   Linux-5.15.0-138-generic-x86_64-with-glibc2.35
Is XPU available:  N/A
DPCPP runtime:     2025.1
MKL version:       2025.1

GPU models and configuration onboard:
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550
* Intel(R) Data Center GPU Max 1550

GPU models and configuration detected:
N/A

Driver version:                                                                                                              [57/1833]
* intel_opencl: 25.13.33276.22-1133~22.04
* level_zero:   N/A

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        52 bits physical, 57 bits virtual
Byte Order:                           Little Endian
CPU(s):                               224
On-line CPU(s) list:                  0-223
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Platinum 8480+
CPU family:                           6
Model:                                143
Thread(s) per core:                   2
Core(s) per socket:                   56
Socket(s):                            2
Stepping:                             6
CPU max MHz:                          3800.0000
CPU min MHz:                          800.0000
BogoMIPS:                             4000.00
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxs
r sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid
aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid dca sse4_1 sse4_2 x2a
pic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb cat_l3 cat_l2 cdp_l3 invpcid_s
ingle intel_ppin cdp_l2 ssbd mba ibrs ibpb stibp ibrs_enhanced tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 a
vx2 smep bmi2 erms invpcid cqm rdt_a avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb intel_pt avx512cd sha_ni avx512bw avx
512vl xsaveopt xsavec xgetbv1 xsaves cqm_llc cqm_occup_llc cqm_mbm_total cqm_mbm_local split_lock_detect avx_vnni avx512_bf16 wbnoinvd
 dtherm ida arat pln pts hwp hwp_act_window hwp_epp hwp_pkg_req avx512vbmi umip pku ospke waitpkg avx512_vbmi2 gfni vaes vpclmulqdq av
x512_vnni avx512_bitalg tme avx512_vpopcntdq la57 rdpid bus_lock_detect cldemote movdiri movdir64b enqcmd fsrm md_clear serialize tsxl
dtrk pconfig arch_lbr amx_bf16 avx512_fp16 amx_tile amx_int8 flush_l1d arch_capabilities
L1d cache:                            5.3 MiB (112 instances)
L1i cache:                            3.5 MiB (112 instances)
L2 cache:                             224 MiB (112 instances)
L3 cache:                             210 MiB (2 instances)
NUMA node(s):                         2
NUMA node0 CPU(s):                    0-55,112-167
NUMA node1 CPU(s):                    56-111,168-223
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl and seccomp
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; B
HI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip] dpcpp-cpp-rt==2025.1.1
[pip] galore-torch==1.0
[pip] impi-rt==2021.15.0
[pip] intel-cmplr-lib-rt==2025.1.1
[pip] intel-cmplr-lib-ur==2025.1.1
[pip] intel-cmplr-lic-rt==2025.1.1
[pip] intel-opencl-rt==2025.1.1
[pip] intel-openmp==2025.1.1
[pip] intel-pti==0.12.3
[pip] intel-sycl-rt==2025.1.1
[pip] mkl==2025.1.0
[pip] mpi4py==4.0.3
[pip] numpy==1.26.4
[pip] oneccl==2021.15.2
[pip] oneccl-devel==2021.15.2
[pip] onemkl-sycl-blas==2025.1.0
[pip] onemkl-sycl-dft==2025.1.0
[pip] onemkl-sycl-lapack==2025.1.0
[pip] onemkl-sycl-rng==2025.1.0
[pip] onemkl-sycl-sparse==2025.1.0
[pip] optimum==1.26.1
[pip] optimum-quanto==0.2.7
[pip] pytorch-msssim==1.0.0
[pip] pytorch-triton-xpu==3.4.0+gitae324eea
[pip] sentence-transformers==4.1.0
[pip] torch==2.8.0.dev20250624+xpu
[pip] torchao==0.12.0+git1c9f2fe
[pip] torchaudio==2.8.0.dev20250624+xpu
[pip] torchdata==0.11.0
[pip] torchvision==0.23.0.dev20250624+xpu
[pip] transformers==4.54.0.dev0

```

**Comment by jiqing-feng** [comment]
Hi @CuiYifeng . Please let me know if you have any updates. Thanks!

**Comment by CuiYifeng** [comment]
@yucai-intel Any new findings?

**Comment by yucai-intel** [comment]
OneDNN conv1d (memory format: acbd) primitive error
On XPU:
onednn_verbose,v1,info,graph,backend,0:dnnl_backend
onednn_verbose,v1,primitive,info,template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,v1,graph,info,template:operation,engine,partition_id,partition_kind,op_names,data_formats,logical_tensors,fpmath_mode,implementation,backend,exec_time
onednn_verbose,v1,primitive,exec,gpu:0,convolution,jit:ir,forward_training,src:f32::blocked:acdb::f0 wei:f32::blocked:acdb::f0 bia:f32::blocked:a::f0 dst:f32::blocked:acdb::f0,attr-scratchpad:user,alg:convolution_direct,mb4_ic512oc512_ih1oh1kh1sh1dh0ph0_iw441ow441kw1sw1dw0pw0,0.25293
 
On CPIU:
 onednn_verbose,v1,info,graph,backend,0:dnnl_backend
onednn_verbose,v1,primitive,info,template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,v1,graph,info,template:operation,engine,partition_id,partition_kind,op_names,data_formats,logical_tensors,fpmath_mode,implementation,backend,exec_time
onednn_verbose,v1,primitive,exec,cpu,reorder,jit:uni,undef,src:f32::blocked:acdb::f0 dst:f32::blocked:Acdb64a::f0,attr-scratchpad:user,,512x512x1x1,2.22485
onednn_verbose,v1,primitive,exec,cpu,convolution,brgconv_1x1:avx512_core,forward_training,src:f32::blocked:acdb::f0 wei:f32:a:blocked:Acdb64a::f0 bia:f32:a:blocked:a::f0 dst:f32::blocked:acdb::f0,attr-scratchpad:user,alg:convolution_direct,mb4_ic512oc512_ih1oh1kh1sh1dh0ph0_iw441ow441kw1sw1dw0pw0,2.08984

**Comment by yao-matrix** [comment]
@chuanqi129 , is this target to PyTorch 2.10? We have several speech models accuracy issues pending on https://github.com/pytorch/pytorch/pull/162944, thx

**Comment by CuiYifeng** [comment]
Hi @yao-matrix @jiqing-feng, fixing PR is merged. Please check with workloads.

**Comment by jiqing-feng** [comment]
I will try when the next nightly torch version is released.

**Comment by chuanqi129** [comment]
@jiqing-feng have you tried the latest nightly? Can we close this issue now?
